python-checks/src/main/java/org/sonar/python/checks/PandasAddMergeParametersCheck.java

```diff
@@ -50,24 +88,39 @@ private static void verifyMergeCallParameters(SubscriptionContext ctx) {
     CallExpression callExpression = (CallExpression) ctx.syntaxNode();
     Optional.ofNullable(callExpression.calleeSymbol())
       .map(Symbol::fullyQualifiedName)
-      .filter(MERGE_METHODS::contains)
-      .ifPresent(fqn -> missingArguments(ctx, callExpression));
+      .filter(METHODS::contains)
+      .ifPresent(fqn -> missingArguments(fqn, ctx, callExpression));
   }
 
-  private static void missingArguments(SubscriptionContext ctx, CallExpression callExpression) {
+  private static void missingArguments(String fullyQualifiedName, SubscriptionContext ctx, CallExpression callExpression) {
     List<String> parameters = new ArrayList<>();
-    if (TreeUtils.argumentByKeyword("how", callExpression.arguments()) == null) {
-      parameters.add("how");
+    if (argumentIsMissing(fullyQualifiedName, Keywords.HOW, callExpression.arguments())) {
+      parameters.add(Keywords.HOW.getKeyword());
     }
-    if (TreeUtils.argumentByKeyword("on", callExpression.arguments()) == null) {
-      parameters.add("on");
+    if (argumentIsMissing(fullyQualifiedName, Keywords.ON, callExpression.arguments())) {
+      parameters.add(Keywords.ON.getKeyword());
     }
-    if (TreeUtils.argumentByKeyword("validate", callExpression.arguments()) == null) {
-      parameters.add("validate");
+    if (argumentIsMissing(fullyQualifiedName, Keywords.VALIDATE, callExpression.arguments())) {
+      parameters.add(Keywords.VALIDATE.getKeyword());
     }
     if (!parameters.isEmpty()) {
-      ctx.addIssue(callExpression, String.format(messages.get(parameters.size() - 1), parameters.toArray()));
+      parameters.add(fullyQualifiedName.substring(fullyQualifiedName.lastIndexOf('.') + 1));
+      ctx.addIssue(callExpression, String.format(MESSAGES.get(parameters.size() - 2), parameters.toArray()));
     }
+  }
+
+  private static boolean argumentIsMissing(String fullyQualfiedName, Keywords keyword, List<Argument> arguments) {
+    return Optional.of(keyword)
+      .map(kw -> TreeUtils.nthArgumentOrKeyword(getPosition(fullyQualfiedName, kw), kw.getKeyword(), arguments))
+      .isEmpty();
+  }
 
+  private static int getPosition(String fullyQualifiedName, Keywords parameter) {
```

Comment:
Nitpick: Maybe we could make the intent of this method clearer with a name like getArgumentPosition.

Reply:
Changed in 65ecde482e55039b470e0fd514887d0a5d32f9a9.